test/test-filter-rewriter.rb

```diff
@@ -110,4 +110,21 @@ def test_under_score
                            ["ref_column.number"]))
     end
   end
+
+  class RegularExpressionTest < self
+    def rewrite(filter)
+      super(filter,
+            :rewrite_regular_expression => true)
+    end
+
+    def test_rewrite_one
+      assert_equal("column1 @ \"value1\" &! column2 @ \"value2\"",
+                   rewrite("column1 @ \"value1\" && column2 @~ \"^(?!.*value2).+$\""))
+    end
+
+    def test_rewrite_multiple
+      assert_equal("column1 @ \"value1\" &! column2 @ \"value2\" &! column2 @ \"value3\" &! column2 @ \"value4\"",
+                   rewrite("column1 @ \"value1\" && column2 @~ \"^(?!.*value2 | value3 | value4).+$\""))
+    end
+  end
 end
```

Comment:
How about non rewrite test case?
I mean that it is  good for rewrite flag is enabled test case, but what about disabled test case?

```ruby
diff --git a/test/test-filter-rewriter.rb b/test/test-filter-rewriter.rb                                                                   
index ea76da8..f501065 100644
--- a/test/test-filter-rewriter.rb
+++ b/test/test-filter-rewriter.rb
@@ -112,9 +112,9 @@ class FilterRewriterTest < Test::Unit::TestCase
   end

   class RegularExpressionTest < self
-    def rewrite(filter)
+    def rewrite(filter, enabled=true)
       super(filter,
-            :rewrite_regular_expression => true)
+            :rewrite_regular_expression => enabled)
     end
 
     def test_rewrite_one
@@ -126,5 +126,11 @@ class FilterRewriterTest < Test::Unit::TestCase
       assert_equal("column1 @ \"value1\" &! column2 @ \"value2\" &! column2 @ \"value3\" &! column2 @ \"value4\"",
                    rewrite("column1 @ \"value1\" && column2 @~ \"^(?!.*value2 | value3 | value4).+$\""))
     end
+
+    def test_no_rewrite
+      assert_equal("column1 @ \"value1\" && column2 @~ \"^(?!.*value2).+$\"",
+                   rewrite("column1 @ \"value1\" && column2 @~ \"^(?!.*value2).+$\"", false))
+    end
+
   end
 end

```